nni/tools/package_utils/common.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+__all__ = ['AlgoMeta']
+
+from typing import Dict, NamedTuple, Optional
+
+class AlgoMeta(NamedTuple):
+    name: str
+    class_name: Optional[str]
+    accept_class_args: bool
+    class_args: Optional[dict]
+    validator_class_name: Optional[str]
+    algo_type: str
```

Comment:
`Literal['tuner', 'assessor', 'advisor']`

Reply:
Literal was introduced in Python 3.8.